pkg/controller/account/account_controller.go

```diff
@@ -158,6 +158,15 @@ func (r *ReconcileAccount) Reconcile(request reconcile.Request) (reconcile.Resul
 		return reconcile.Result{}, err
 	}
 
+	// Add finalizer to non-sts account cr
+	if !currentAcctInstance.Spec.ManualSTSMode {
+		err := r.addFinalizer(reqLogger, currentAcctInstance)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		return reconcile.Result{}, nil
```

Comment:
Wouldn't this just loop at this point forever for non-sts clusters?

Reply:
no? return reconcile.Result{}, nil should break the loop

Reply:
But it would still never get to the next part of the code where it starts building the account, right?  Because it would always hit that return statement?

Reply:
aah gonna delete that last return